### PR TITLE
gup: 0.7.0 -> 0.8.0

### PR DESCRIPTION
**Pull request by timbertson**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 126059` at e3cf1c63 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>gup</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/tools/build-managers/gup/default.nix:17:3:
```
   |
17 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/development/tools/build-managers/gup/default.nix:16:3:
```
   |
16 |   buildPhase = "make python";
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/development/tools/build-managers/gup/default.nix [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 , which, runtimeShell, pychecker ? null }:
 stdenv.mkDerivation rec {
-  version = "0.7.0";
+  version = "0.8.0";
   src = fetchFromGitHub {
     owner = "timbertson";
     repo = "gup";
-    rev = "version-0.7.0";
-    sha256 = "1pwnmlq2pgkkln9sgz4wlb9dqlqw83bkf105qljnlvggc21zm3pv";
+    rev = "version-0.8.0";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/build-managers/gup/default.nix:8]:
```suggestion
    rev = "version-${version}";
```

File: pkgs/development/tools/build-managers/gup/default.nix [hunk 2]
```diff
@@ -18,20 +18,23 @@ stdenv.mkDerivation rec {
...
+      ${nix-update-source}/bin/nix-update-source \
+        --prompt version \
+        --replace-attr version \
+        --set owner timbertson \
+        --set repo gup \
+        --set type fetchFromGitHub \
+        --set rev 'version-{version}' \
+        --modify-nix default.nix
+    ''
+  ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/build-managers/gup/default.nix:37]: This can most likely be replaced with nix-update.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/build-managers/gup/default.nix (hunk 2 above)
Comment (timbertson) [reply pkgs/development/tools/build-managers/gup/default.nix:37]: That sounds good, this update script is creaking at the seams. But that will need more work I'm afraid

**Comment by timbertson** [comment]
This version fixes an annoying bug on darwin, can changes to the update script be deferred for later @SuperSandro2000 ? They won't be trivial as I've never used `nix-update` before.

**Comment by timbertson** [comment]
Also I just kicked the tyres, and it looks like there's at least two bugs, the bad version number detection and an unknown issue with `nix-prefetch`:

```
$ nix-update gup
( ... )
fetch https://github.com/timbertson/gup/releases.atom
Update 0.8.0 -> ersion-0.8.0 in /Users/tcuthbertson/Code/nix/nixpkgs/pkgs/development/tools/build-managers/gup/default.nix
$ nix-prefetch (import ./. (if (builtins.hasAttr "config" (builtins.functionArgs (import ./.))) then { config.checkMeta = false; } else { })).gup
error: Unsupported input type ''
```

..and `nix-prefetch --help` just says `No manual entry for nix-prefetch`

😞

**Comment by SuperSandro2000** [comment]
I thought this smoke just work but if it doesn't we can fix this another time.

**Pull request merged**